src/main/java/seedu/nuscents/data/TransactionList.java

```diff
@@ -0,0 +1,91 @@
+package seedu.nuscents.data;
+
+import seedu.nuscents.data.transaction.Transaction;
+import seedu.nuscents.data.transaction.TransactionCategory;
+import seedu.nuscents.ui.Ui;
+
+import java.util.ArrayList;
+
+import static seedu.nuscents.ui.Messages.LINE;
+
+public class TransactionList {
+    private ArrayList<Transaction> transactions;
+
+    public TransactionList() {
+        transactions = new ArrayList<Transaction>();
+    }
+
+    public TransactionList(ArrayList<Transaction> transactions) {
+        this.transactions = transactions;
+    }
+
+    public ArrayList<Transaction> getTransactions() {
+        return transactions;
+    }
+
+    public void addTransaction(Transaction transaction) {
+        transactions.add(transaction);
+        Ui.showTransactionAddedMessage(transaction);
+    }
+
+    public void deleteTransaction(int transactionIndex) {
+        Transaction transaction = transactions.get(transactionIndex-1);
+        transactions.remove(transaction);
+        Transaction.decreaseTransactionCountByOne();
+        Ui.showTransactionRemovedMessage(transaction);
+    }
+
+    public void findTask(String keyword) {
+        ArrayList<Transaction> results = new ArrayList<>();
+        boolean isFound = false;
+        for (Transaction transaction : transactions) {
+            if (transaction.getDescription().contains(keyword.toLowerCase())) {
+                isFound = true;
+                results.add(transaction);
+            }
+        }
+        System.out.println(LINE);
+        if (isFound) {
+            int resultCount = 1;
+            System.out.println("Here are the matching tasks in your list:");
+            for (Transaction transaction : results) {
+                System.out.println(resultCount + ". " + transaction.getDescription());
+            }
+        } else {
+            System.out.println("No matching tasks are found :/");
+        }
+        System.out.println(LINE);
+    }
+
+    public void viewTransaction(int transactionIndex) {
+        Transaction transaction = transactions.get(transactionIndex-1);
+        Ui.showTransactionViewMessage(transaction);
+    }
+
+    public void filterTransaction(TransactionCategory category) {
+        ArrayList<Transaction> filteredTransactions = new ArrayList<>();
+        boolean isFound = false;
+
+        for (Transaction transaction : transactions) {
+            if (transaction.getCategory() == category) {
+                filteredTransactions.add(transaction);
+                isFound = true;
+            }
+        }
+
+        if (isFound) {
+            Ui.showFilterMessage(filteredTransactions, category);
+            System.out.println("Filtered transactions in the category " + category + ":");
```

Comment:
Perhaps your system output here can be moved to the UI class? This ensures each class has follows Single Responsibility Principle e.g. printing to the console is handled solely by the UI class.